frontend/src/layout/header/UserBox.tsx

```diff
@@ -107,149 +171,189 @@ const WithFloatingMenu: React.FC<{ children: ReactElement }> = ({ children }) =>
-                            display: "none",
+                            borderRadius: "8px 8px 0 0",
                         },
                     }}
-                >{t("user.login")}</MenuItem>
-            )}
-
-            {isRealUser(user) && <>
+                    icon={<FiFolder />}
+                    borderBottom
+                    linkTo="/~manage"
+                    onClick={() => close()}
+                >{t("user.manage-content")}</MenuItem>
                 {user.canUpload && <MenuItem
                     icon={<FiUpload />}
                     linkTo={"/~upload"}
                     onClick={() => close()}
                 >{t("upload.title")}</MenuItem>}
-                <MenuItem
-                    icon={<FiFilm />}
-                    linkTo="/~manage"
-                    onClick={() => close()}
-                >{t("user.manage-content")}</MenuItem>
             </>}
 
-            <MenuItem icon={<HiOutlineTranslate />} onClick={() => setState("language")}>
-                {t("language")}
-            </MenuItem>
-
             {/* Logout button if the user is logged in */}
             {isRealUser(user) && <Logout />}
         </>,
         language: () => <>
-            <MenuItem icon={<FiChevronLeft />} onClick={() => setState("main")} borderBottom>
-                {t("back")}
-            </MenuItem>
-            <LanguageMenu />
+            <ReturnButton onClick={() => close()}>{t("Choose language")}</ReturnButton>
+            <LanguageMenu close={() => close()} />
         </>,
     });
 
-    return (
-        <Floating padding={0}>
+    return <Floating borderWidth={0} padding={0}>
+        <div
+            onClick={e => {
+                if (e.target === e.currentTarget) {
+                    close();
+                }
+            }}
+            onBlur={handleBlur}
+            css={{
+                position: "relative",
+                // Grey out background on mobile devices.
+                [`@media (max-width: ${BREAKPOINT_MEDIUM}px)`]: {
+                    position: "fixed",
+                    top: "var(--header-height)",
+                    bottom: 0,
+                    left: 0,
+                    right: 0,
+                    zIndex: 1001,
+                    backgroundColor: "#000000a0",
+                },
+            }}>
             <ul css={{
-                listStyle: "none",
+                borderRadius: 8,
+                zIndex: 1000,
+                right: 0,
                 margin: 0,
-                paddingLeft: 0,
                 minWidth: 200,
+                paddingLeft: 0,
+                overflow: "hidden",
+                listStyle: "none",
+                [`@media (max-width: ${BREAKPOINT_MEDIUM}px)`]: {
+                    backgroundColor: "white",
+                    marginTop: 0,
+                    position: "fixed",
+                    left: 0,
+                    top: 0,
+                    boxShadow: "none",
```

Comment:
I don't think this does anything? What other `boxShadow` declaration would it overwrite?